enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/protocol/handshake/ClientMessageDecoder.java

```diff
@@ -88,4 +79,22 @@ protected void decode( ChannelHandlerContext ctx, ByteBuf in, List<Object> out )
             return;
         }
     }
+
+    private <T extends BaseProtocolResponse> Optional<T> decodeProtocolResponse( TriFunction<StatusCode, String, Integer, T> constructor, ByteBuf in )
```

Comment:
Optional here doesn't seem good. We want to fail and throw an exception if there is some sort of protocol violation. Decoding completely and correctly is not optional.

Reply:
Yes, this is to have something working before implementing handling an unknown status code. There is a TODO below.